frontend/javascripts/oxalis/model/actions/volumetracing_actions.ts

```diff
@@ -126,11 +126,16 @@ export const initializeEditableMappingAction = (mapping: ServerEditableMapping)
  * has dealt with the case where the maximum segment id is not set. In that case,
  * the create cell action should not be exposed via the UI.
  */
-export const createCellAction = (largestSegmentId: number) =>
-  ({
+export const createCellAction = (activeCellId: number, largestSegmentId: number) => {
+  // The largestSegmentId is only updated if a voxel using that id was annotated. Therefore, it can happen
+  // that the activeCellId is larger than the largestSegmentId. Choose the larger of the two ids increased by one.
+  const newSegmentId =
+    largestSegmentId && largestSegmentId > activeCellId ? largestSegmentId + 1 : activeCellId + 1;
+  return {
     type: "CREATE_CELL",
-    largestSegmentId,
-  } as const);
+    newSegmentId,
+  } as const;
+};
 
```

Comment:
Here I followed your suggestion to add the desired updated `activeCellId` value to the action. This required pulling the logic of choosing between `largestSegmentId` and `activeCellId` out of the reducer and frontloading it to the code that creates the action. Or do you see another option to not frontload the logic but still be able to add the `newSegmentId` parameter to the action?

I am not perfectly happy with putting the logic of choosing the larger value here. But otherwise, the logic would need to appear before every call to the `createCellAction` action function and thus would have more code duplication. Thus I would say that this is the better solution under the given circumstances.

Reply:
I think your current way is currently the best trade-off :+1: